packages/backend-api/integration/integrate.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+INTEGRATION_DIR="$( cd "$(dirname "$0")" ; pwd -P )"
+PROJECT_BASE_PATH=`[ -d "$PWD/project-base" ] && echo "$PWD/project-base" || echo "$PWD"`
+
+## TODO check that the ${PROJECT_BASE_PATH} is correct project-base
+
+echo "Copying FOS REST configuration..."
+cp ${INTEGRATION_DIR}/app/config/packages/fos_rest.yml ${PROJECT_BASE_PATH}/app/config/packages/fos_rest.yml || exit 1
+echo "Done"
+
+echo "Copying OAuth2 configuration..."
+cp ${INTEGRATION_DIR}/app/config/packages/trikoder_oauth2.yml ${PROJECT_BASE_PATH}/app/config/packages/trikoder_oauth2.yml || exit 1
+mkdir -p ${PROJECT_BASE_PATH}/app/config/packages/oauth2 || exit 1
+cp ${INTEGRATION_DIR}/app/config/packages/oauth2/.gitignore ${PROJECT_BASE_PATH}/app/config/packages/oauth2/.gitignore || exit 1
+cp ${INTEGRATION_DIR}/app/config/packages/oauth2/parameters_oauth.yml.dist ${PROJECT_BASE_PATH}/app/config/packages/oauth2/parameters_oauth.yml.dist || exit 1
+echo "Done"
+
+echo "Creating directory src/Shopsys/ShopBundle/Controller/Api/V1..."
+mkdir -p ${PROJECT_BASE_PATH}/src/Shopsys/ShopBundle/Controller/Api/V1
+touch ${PROJECT_BASE_PATH}/src/Shopsys/ShopBundle/Controller/Api/V1/.gitkeep
+echo "Done"
+
+echo "Copying tests..."
+cp ${INTEGRATION_DIR}/tests/ShopBundle/Smoke/BackendApiTest.php ${PROJECT_BASE_PATH}/tests/ShopBundle/Smoke/BackendApiTest.php || exit 1
+cp ${INTEGRATION_DIR}/tests/ShopBundle/Test/OauthTestCase.php ${PROJECT_BASE_PATH}/tests/ShopBundle/Test/OauthTestCase.php || exit 1
+echo "Done"
+
+function apply_patch () {
+    if [ -z $1 ]
+    then
+        echo "Please provide path as first parameter of apply_patch function"
+        exit 0
+    fi
+    
+    local FILE_PATH=$1
+    local SOURCE_FILE_PATH=$1
+
+    if [ ! -z $2 ]
+    then
+        SOURCE_FILE_PATH=$2
+    fi
+
+    echo "Applying patch for ${FILE_PATH}..."
+    local PATCH_DRY_RUN=`patch -st --dry-run ${PROJECT_BASE_PATH}/${FILE_PATH} ${INTEGRATION_DIR}/${SOURCE_FILE_PATH}.patch`
+    local PATCH_REVERSED=`grep "Reversed" <<< ${PATCH_DRY_RUN}`
```

Comment:
maybe we can use just return values. or could be there more result types than this?
and maybe `-e` will cover the whole thing

Reply:
There are 3 states after the `dry run`
* no error -> will be patched on line 58
* error -> the script stops
* the `patch` detects the patch is already applied. In such case we are skipping the patch and continue with the script

So I would say we cannot use return values. Also the "Reverse" returns code `0`